WordPress/Sniffs/PHP/DiscouragedPHPFunctionsSniff.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Discourages the use of various native PHP functions and suggests alternatives.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.11.0
+ */
+class WordPress_Sniffs_PHP_DiscouragedPHPFunctionsSniff extends WordPress_AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Groups of functions to discourage.
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type'      => 'error' | 'warning',
+	 * 		'message'   => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'create_function' => array(
+				'type'      => 'warning',
+				'message'   => '%s() is discouraged, please use anonymous functions instead.',
+				'functions' => array(
+					'create_function',
+				),
+			),
+
+			'serialize' => array(
+				'type'      => 'warning',
+				'message'   => '%s() found. Serialized data has known vulnerability problems with Object Injection. JSON is generally a better approach for serializing data. See https://www.owasp.org/index.php/PHP_Object_Injection',
+				'functions' => array(
+					'serialize',
+					'unserialize',
+				),
+			),
+
+			'urlencode' => array(
+				'type'      => 'warning',
+				'message'   => '%s() should only be used when dealing with legacy applications rawurlencode() should now be used instead. See http://php.net/manual/en/function.rawurlencode.php and http://www.faqs.org/rfcs/rfc3986.html',
+				'functions' => array(
+					'urlencode',
+				),
+			),
+
+			'runtime_configuration' => array(
+				'type'      => 'warning',
+				'message'   => '%s() found. Changing configuration at runtime should not be done.',
+				'functions' => array(
+					'error_reporting',
+					'ini_alter',
+					'ini_restore',
+					'ini_set',
+					'apache_setenv',
+					'putenv',
+					'set_include_path',
+					'restore_include_path',
+					// This alias was DEPRECATED in PHP 5.3.0, and REMOVED as of PHP 7.0.0.
```

Comment:
They are included because they touch the runtime. For PHP compatibility we can use the PHPcompatibility sniffs.